### DISCUSSION TITLE
"keys" feature introduced in 3.1.0 is under Master Software License Agreement

### DISCUSSION THREAD

**Discussion by pantowsk** [Kong/kong#10406] [Help]
Files revealed to "keys" feature introduced in 3.1.0: 
- Add keys entity to store and manage asymmetric keys. https://github.com/Kong/kong/pull/9737
- Add key-sets entity to group and manage keys https://github.com/Kong/kong/pull/9737

kong/db/schema/entities/keys.lua
kong/db/schema/entities/key_sets.lua
kong/db/dao/key_sets.lua
kong/db/dao/keys.lua
and few others are under Enterprise Software License Agreement:

_-- This software is copyright Kong Inc. and its licensors.
-- Use of the software is subject to the agreement between your organization
-- and Kong Inc. If there is no such agreement, use is governed by and
-- subject to the terms of the Kong Master Software License Agreement found
-- at https://konghq.com/enterprisesoftwarelicense/.
-- [ END OF LICENSE 0867164ffc95e54f04670b5169c09574bdbd9bba ]_

Is it intention to publish under Enterprise licensed feature within Apache 2.0 license package

**Comment by jschmid1** [comment]
Nice catch @pantowsk. We'll take care of resolving this. Thanks for reporting!

**Reply by jschmid1** [reply]
Fixed in https://github.com/Kong/kong/pull/10571

**Comment by pantowsk** [comment]
Only 4 out of 7 licensed files have been changed. The following files from https://github.com/Kong/kong/pull/9737 are under Enterprise Software License Agreement:

spec/02-integration/03-db/19-key-sets_spec.lua
spec/02-integration/03-db/18-keys_spec.lua
spec/02-integration/04-admin_api/21-admin-api-keys_spec.lua

and recently modified:
spec/fixtures/custom_plugins/kong/plugins/dummy/schema.lua
